python/django_bolt/openapi/schema_generator.py

```
@@ -1321,6 +1348,33 @@ def _struct_to_schema(self, struct_type: type) -> Schema:
+        instead of producing a silently wrong spec.
+        """
+        existing = self._component_types.get(schema_name)
+        if existing is None:
+            self._component_types[schema_name] = cls
+            return True
+        if existing is not cls:
+            raise ValueError(
+                f"OpenAPI component name collision: {cls!r} and {existing!r} "
+                f"both map to component schema '{schema_name}'. Rename one type "
+                f"so each component has a unique name."
+            )
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)